File: Ejemplo.kt
fun sumar(a: Int, b: Int): Int {
    return a + b
}

fun main() {
    print("Ingrese su nombre: ")
    val nombre = readLine()

    print("Ingrese el primer numero: ")
    val num1 = readLine()!!.toInt()

    print("Ingrese el segundo numero: ")
    val num2 = readLine()!!.toInt()

    val resultado = sumar(num1, num2)

    println("Hola $nombre")
    println("La suma es: $resultado")
}

File: Ejemplo.scala
object Main {

  def sumar(a: Int, b: Int): Int = {
    a + b
  }

  def main(args: Array[String]): Unit = {

    print("Ingrese su nombre: ")
    val nombre = scala.io.StdIn.readLine()

    print("Ingrese el primer numero: ")
    val num1 = scala.io.StdIn.readInt()

    print("Ingrese el segundo numero: ")
    val num2 = scala.io.StdIn.readInt()

    val resultado = sumar(num1, num2)

    println(s"Hola $nombre")
    println(s"La suma es: $resultado")
  }
}
